src/adapter.jl

```diff
@@ -271,3 +271,29 @@ Base.hash(::FFE, h::UInt) = h
 # only if `x` is a multiplicative element in the sense of GAP.
 Base.literal_pow(::typeof(^), x::GapObj, ::Val{-1}) = Globals.INV_MUT(x)
 
+# iteration
+
+function Base.iterate(obj::GapObj)
+    islist = Globals.IsList(obj)
+    islist || Globals.IsCollection(obj) ||
+        throw(ArgumentError("object cannot be iterated"))
+    if islist
+        iterate(obj, (1, Globals.Length(obj)))
+    else
+        iterate(obj, Globals.Iterator(obj))
+    end
```

Comment:
Seems a bit artificial to force the error to be at the start... how about this?
```suggestion
    if Globals.IsList(obj)
        iterate(obj, (1, Globals.Length(obj)))
    elseif Globals.IsCollection(obj)
        iterate(obj, Globals.Iterator(obj))
    else
        throw(ArgumentError("object cannot be iterated"))
    end
```